userland/libnrfserialization/NRFMakefile.mk

```diff
@@ -0,0 +1,310 @@
+SOURCE_PATHS += $(wildcard $(SDK_SOURCE_PATH)serialization/common/)
+SOURCE_PATHS += $(wildcard $(SDK_SOURCE_PATH)serialization/common/transport/ser_phy/)
+SOURCE_PATHS += $(wildcard $(SDK_SOURCE_PATH)serialization/common/transport/ser_phy/config/)
+
+
+CFLAGS += -DSVCALL_AS_NORMAL_FUNCTION -DBLE_STACK_SUPPORT_REQD
+
+# How many central/peripherals are defined changes how much memory the
+# softdevice requires. Change the amount of memory allotted in a custom ld
+# file if your configuration is different than default.
+
+ifeq ($(RAM_KB), 16)
+  # limit 16 kB RAM nRFs to only act as peripherals. Doing otherwise
+  # requires careful balancing of memory requirements and should be done
+  # manually, not automatically
+  CENTRAL_LINK_COUNT ?= 0
+  PERIPHERAL_LINK_COUNT ?= 1
+else
+  CENTRAL_LINK_COUNT ?= 1
+  PERIPHERAL_LINK_COUNT ?= 1
+endif
+
+LIBRARY_PATHS += $(wildcard $(SDK_INCLUDE_PATH)ble/*/)
+LIBRARY_PATHS += $(wildcard $(SDK_INCLUDE_PATH)ble/ble_services/*/)
+SOURCE_PATHS += $(wildcard $(SDK_SOURCE_PATH)ble/*/)
+SOURCE_PATHS += $(wildcard $(SDK_SOURCE_PATH)ble/ble_services/*/)
+CFLAGS += -DBLE_STACK_SUPPORT_REQD -DSOFTDEVICE_PRESENT -DCENTRAL_LINK_COUNT=$(CENTRAL_LINK_COUNT) -DPERIPHERAL_LINK_COUNT=$(PERIPHERAL_LINK_COUNT)
+
+ifdef ENABLE_WIRELESS_DFU
+    CFLAGS += -DENABLE_DFU
+    APPLICATION_SRCS += bootloader_util.c
+endif
+
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)softdevice/common/softdevice_handler/
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)softdevice/$(SOFTDEVICE_MODEL)/headers/
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)softdevice/$(SOFTDEVICE_MODEL)/headers/nrf51
+
+SOURCE_PATHS += $(SDK_SOURCE_PATH)softdevice/common/softdevice_handler/
+SOURCE_PATHS += $(SDK_SOURCE_PATH)softdevice/$(SOFTDEVICE_MODEL)/headers/
+SOURCE_PATHS += $(SDK_SOURCE_PATH)softdevice/$(SOFTDEVICE_MODEL)/headers/nrf51
+
+# Load the sources from the serialization library
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)serialization/$(SERIALIZATION_MODE)/codecs/$(SOFTDEVICE_MODEL)/middleware/
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)serialization/$(SERIALIZATION_MODE)/codecs/$(SOFTDEVICE_MODEL)/serializers/
+LIBRARY_PATHS += $(SDK_INCLUDE_PATH)serialization/common/struct_ser/$(SOFTDEVICE_MODEL)/
+
+SOURCE_PATHS += $(SDK_SOURCE_PATH)serialization/$(SERIALIZATION_MODE)/codecs/$(SOFTDEVICE_MODEL)/middleware/
+SOURCE_PATHS += $(SDK_SOURCE_PATH)serialization/$(SERIALIZATION_MODE)/codecs/$(SOFTDEVICE_MODEL)/serializers/
+SOURCE_PATHS += $(SDK_SOURCE_PATH)serialization/common/struct_ser/$(SOFTDEVICE_MODEL)/
+
+
+
+print-% : ; @echo $* = $($*)
+
+LIBRARY_INCLUDES = $(addprefix -I,$(LIBRARY_PATHS))
+CMSIS_INCLUDE = $(addprefix -I,$(CMSIS_INCLUDE_PATH))
+
+VPATH = $(SOURCE_PATHS)
+
+CFLAGS += -g -D$(DEVICE) -D$(BOARD) $(LIBRARY_INCLUDES) -std=c11 -Os -DSDK_VERSION_$(SDK_VERSION) -DSOFTDEVICE_$(SOFTDEVICE_MODEL)
+CFLAGS += -D$(shell echo $(SOFTDEVICE_MODEL) | tr a-z A-Z)
+CFLAGS += -D__TOCK__
+COMPILE_ONLY += -c
+
+# Needs to be done per-arch by our Makefile
```

Comment:
Those are the "stock" rules from NRF land that we replaced, left as comments for someone comparing this in the future